### PR TITLE
feat: added a new flag that displays the number of tokens sent in prompt and received in completion

### LINKED ISSUE #12: Feature: Chat completion token count
# Description

I think it would be really nice to have a feature which tells the user how many tokens were sent in the prompt, and how many tokens were returned in the chat completion. Something like `--token-usage` or `-t`.

# Usage

one would use `genereadme` just like they normally would, except with passing the new argument, i.e.:

```bash
genereadme main.cpp --token-usage
```
or
```bash
genereadme main.cpp -t
```

And the results would be the exact same in addition to the following:
```
Prompt token count: NUMBER_OF_TOKENS
Completion token count: NUMBER_OF_TOKENS
```

### PR DESCRIPTION
**Pull request by Mounayer**
# Description

Closes #12.

- Added a new flag `--token-usage` which when given, prints the number of tokens that were sent in the `prompt` and the number of tokens that were returned in the `completion` to `stderr.

This simply required the addition for another flag check `--token-usage`:

```javascript
   .option("--token-usage", "Show prompt and completion token usage")
```

I've also made sure to keep your naming conventions/formatting style consistent, in the for loop that does the chat completion for each file processed, I have accumulated the total tokens `sent` and received:

```javascript
    promptTokens += response.usage.prompt_tokens;
    completionTokens += response.usage.completion_tokens;
```

which I then display at the end of program run-time if the `--token-usage` flag is provided as such:

```javascript
    if (program.opts().tokenUsage) {
      console.error(`Prompt tokens: ${promptTokens}`);
      console.error(`Completion tokens: ${completionTokens}`);
    }
```

- Updated `README.md` to explain the new flag.

### Test 1

```bash
genereadme examples/sum.js --token-usage
```

This should display something like:

image

### Test 2

You can try it out with multiple files too, i.e.:

```bash
genereadme examples/sum.js examples/createUser.js --token-usage
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ program
     "-t, --temperature <temperature>",
     "Temperature for the chat completions"
   )
+  .option("--token-usage", "Show prompt and completion token usage")
```
Comment (cleobnvntra) [inline index.js:32]: We can use the shorthand `-tu` flag for this since `-t` is already used for temperature!

**Review by cleobnvntra — CHANGES_REQUESTED** [review]
Looks great! Code looks like it was written by me!
Just had a suggestion for the shorthand flag.

File: index.js (hunk 1 above)
Comment (Mounayer) [reply index.js]: That's a great idea! Updated

**Review by cleobnvntra — APPROVED** [review]
Looks great!

**Pull request merged**